### DISCUSSION TITLE
Welcome to BananaWRT Discussions!

### DISCUSSION THREAD

**Discussion by SuperKali** [SuperKali/BananaWRT#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by pijwarren** [comment]
Any chance you could do a build for the Bpi-R4? I've been fighting against my fm350 to get it working and had no luck so far

**Reply by SuperKali** [reply]
Hi,

I suggest you create a separate discussion next time.

Regarding the support of BPI R4, yes I plan to add the support, but it will be possible as soon as the latter is at least stable, owning that board I have already carried out various tests and there are many things that do not work.

So I advise you to be patient and follow the development of the project.

**Comment by darkwomb** [comment]
Hello, 

Just installed BananaWRT on r3 mini and thank you very much Dany for amazing work. 

R3 mini will be a main modem / router for my home as my old ZTE MF286 LTE modem seems a bit outdated ( im actually planning to install openwrt on it so that it will be as a hub for all my esphome devices in home assistant ).

Getting back to BananaWRT on r3 mini, I'm using FM350-GL and it is HP model. Currently waiting for m.2 to usb3.0 adapter so that i can flash it to a newer version. Tried to plug it in to wwan slot on my laptop but it doesn't support flashing as i believe the laptop port is actually pcie. Current Fm350-gl version 19.17.10. However it works , i can browse internet and speedtest is showing around 40mbps DL and 40mbps UP but there is no information regarding the cell tower, band or signal when checking status in BananaWRT.

And here is my question, I'm hoping someone could answer. 

1. My nearest tower is currently 4G LTE only with operators future plans to upgrade to 5G and im getting 240mbps DL and 50mbps UP when using phone with same sim card. Im seeing low DL and UP with fm350-gl because it's on old firmware? 19.17.10

Thank you!

**Reply by SuperKali** [reply]
Hi! Thank you so much for the kind words about BananaWRT - it's always a pleasure to know that the work is appreciated and that it's working well for you! 😊

### The problem with firmware 19.17.10

You've hit the nail on the head: the version 19.17.10 you have installed is really too old. The main issue is that this version doesn't support dual AT Commands. What happens is that only ttyUSB3 is exposed for sending and receiving AT commands.
With the newer firmware updates, ttyUSB1 is introduced, which allows you to send AT commands separately, and consequently you'll finally be able to see the current network parameters that you're missing now (information about cell tower, band, signal, etc.).

### Regarding performance

As for the speed problem, not knowing the current network parameters makes it difficult for me to understand the exact reason for the slowness. However, I can be certain that many users are using the Fibocom FM350 with excellent results, and 5G speeds are typically around 600-700 Mbps, depending obviously on the BTS you're connecting to.
The difference with your phone's 240 Mbps could be due precisely to the combination of old firmware + lack of network parameter information to optimize the connection.

**My advice:** definitely update the firmware as soon as you receive the M.2 to USB 3.0 adapter. You'll immediately see the difference both in available information and performance!
